src/python/WMCore/ReqMgr/DataStructs/Request.py

```diff
@@ -84,6 +79,30 @@ def initialize_resubmission(request_args, reqmgr_db_service):
     # to be used later on for spec validation
     request_args["OriginalRequestType"] = parent_args["RequestType"]
 
+def _replace_cloned_args(clone_args, user_args):
+    """
+    replace original arguments with user passed arguement.
+    If the value is dictionary format, overwrite only specified arguments.
+    If the original arguemnt has simple value and user passes dictionary, completely replace to dictionary
+    """
+    for prop in user_args:
+        if isinstance(prop, dict) and isinstance(clone_args.get(prop), dict):
+            _replace_cloned_args(clone_args.get(prop, {}), prop)
+        else:     
+            clone_args[prop] = user_args[prop]
+    return
+    
+def initialize_clone(request_args, reqmgr_db_service):
+    """
+    Initialize a Clone arguments by inheriting and overwriting argument from OriginalRequest
+    """
+    requests = reqmgr_db_service.getRequestByNames(request_args["OriginalRequestName"])
+    clone_args = requests.values()[0]
+    # TODO: need to validate new overwrite request_args here since it will skip the clone_args validtaiton
```

Comment:
Validating only user_args makes sense to me. I'm just worried about legacy workflows (let's say previous release) that could have a data format/etc different that what we expect now. Though this would be a case in 10000 I guess...